silkworm/rpc/http/connection.cpp

```diff
@@ -265,8 +268,22 @@ Task<void> Connection::do_write(const std::string& content, boost::beast::http::
         res.set(boost::beast::http::field::date, get_date_time());
         res.erase(boost::beast::http::field::host);
         res.keep_alive(request_keep_alive_);
-        res.content_length(content.size());
-        res.body() = content;
+        if (compress) {
+            const std::string compression_type = "gzip";
+            res.set(boost::beast::http::field::content_encoding, compression_type);
+            std::string compressed_data;
+            try {
+                compress_data(content, compressed_data);
+            } catch (const std::exception& e) {
+                SILK_ERROR << "Connection::compress_data exception: " << e.what();
+                throw;
```

Comment:
Not sure we do really want to raise an exception here, probably we could directly return an `Internal server error` code or something similar